### PR TITLE
Cleanup dns

### PR DESCRIPTION
**Pull request by bowei**
* Remove hostname endpoints annotation (was beta feature)
* Remove references to non-client-go API
* Replaces references to internal kubernetes API with client-go.

### CODE REVIEW

**Comment by bowei** [comment]
@thockin @MrHohn

File: cmd/kube-dns/app/server.go [hunk 1]
```diff
@@ -69,41 +69,28 @@ func NewKubeDNSServerDefault(config *options.KubeDNSConfig) *KubeDNSServer {
...
-	if dnsConfig.KubeMasterURL != "" && dnsConfig.KubeConfigFile == "" {
-		// Only --kube-master-url was provided.
-		config = &restclient.Config{
-			Host:          dnsConfig.KubeMasterURL,
-			ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Version: "v1"}},
+func newKubeClient(dnsConfig *options.KubeDNSConfig) (kubernetes.Interface, error) {
+	var config *rest.Config
+	var err error
+
+	if dnsConfig.KubeConfigFile != "" {
```
Comment (thockin) [inline cmd/kube-dns/app/server.go]: Can you flip these so InClusterConfig is the first case - that's the normal case.

File: pkg/dns/dns_test.go [hunk 2]
```diff
@@ -207,21 +207,9 @@ func TestSkyPodHostnameSRVLookup(t *testing.T) {
 
 	service := newHeadlessService()
 	endpointIPs := []string{"10.0.0.1", "10.0.0.2"}
-	endpoints := newEndpoints(service, newSubsetWithOnePort("", 80, endpointIPs...))
-
-	// The format of thes annotations is:
-	// endpoints.beta.kubernetes.io/hostnames-map: '{"ep-ip":{"HostName":"pod request hostname"}}'
-	epRecords := map[string]endpointsapi.HostRecord{}
-	for i, ep := range endpointIPs {
-		epRecords[ep] = endpointsapi.HostRecord{HostName: fmt.Sprintf("ep-%d", i)}
```
Comment (thockin) [inline pkg/dns/dns_test.go]: can you nix the HostRecord struct, or is that still in use?

**Review by thockin — COMMENTED** [review]
nice

File: pkg/dns/dns_test.go (hunk 2 above)
Comment (bowei) [inline pkg/dns/dns_test.go]: confused about this comment -- it is deleted in this PR?

File: cmd/kube-dns/app/server.go (hunk 1 above)
Comment (bowei) [inline cmd/kube-dns/app/server.go]: done

File: pkg/dns/dns_test.go (hunk 2 above)
Comment (MrHohn) [inline pkg/dns/dns_test.go]: I guess Tim means https://github.com/kubernetes/kubernetes/blob/master/pkg/api/endpoints/util.go#L38-L41 and https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/endpoints/util.go#L38-L41.

Comment (bowei) [inline pkg/dns/dns_test.go]: I'm going to do this in a follow on PR, as it touches endpoint controller

**Comment by bowei** [comment]
updated

**Comment by bowei** [comment]
@thockin @MrHohn -- review updated

File: pkg/dns/dns_test.go (hunk 2 above)
Comment (thockin) [inline pkg/dns/dns_test.go]: OK

**Comment by thockin** [comment]
LGTM - @MrHohn

**Pull request merged**